### PR TITLE
raftstore/error: add StoreNotMatch details

### PR DESCRIPTION
**Pull request by BusyJay**
It helps debugging implementation errors.

Should not merge this before pingcap/kvproto#333.

### CODE REVIEW

File: src/raftstore/errors.rs [hunk 1]
```diff
@@ -166,7 +166,14 @@ impl Into<errorpb::Error> for Error {
                     .mut_raft_entry_too_large()
                     .set_entry_size(entry_size);
             }
-            Error::StoreNotMatch(..) => errorpb.set_store_not_match(errorpb::StoreNotMatch::new()),
+            Error::StoreNotMatch(to_store_id, my_store_id) => {
+                errorpb
+                    .mut_store_not_match()
+                    .set_expect_store_id(to_store_id);
```
Comment (disksing) [inline src/raftstore/errors.rs]: Should `expect_store_id` be `my_store_id`?

Comment (BusyJay) [reply src/raftstore/errors.rs]: How about `requested_store_id` and `actual_store_id`?

Comment (BusyJay) [reply src/raftstore/errors.rs]: Or `expect_store_id` and `actual_store_id`?

Comment (ngaut) [reply src/raftstore/errors.rs]: expect_store_id and actual_store_id +1

Comment (disksing) [reply src/raftstore/errors.rs]: I think `request_store_id` is better. By `expect`, I connot know it is the client or the server that is expecting.

**Pull request merged**